File: listPrime.cpp
#include <iostream>
#include <math.h>

bool isPrime(int n, int *primes) {
	int index = 0;
	while (primes[index] != 0 && primes[index] < sqrt(n)) {
		if (n % primes[index] == 0) {
			return false;
		}
		index++;
	}
	return true;
}

void listPrime(int max) {
	int *primes = (int*)malloc(sizeof(int)*max);
	primes[0] = 2;
	int index = 1;
	for (int i = 2; i <= max; i++) {
		if (isPrime(i, primes)) {
			std::cout << i << std::endl;
			primes[index++] = i;
		}
	}
	free(primes);
}

int main(int argc, char **argv) {
	int max;
	if (argc == 1) {
		std::cout << "Enter max: ";
		std::cin >> max;
	} else { max = atoi(argv[1]); }
	if (max == 0) { max = 4096; }

	listPrime(max);

	return 0;
}

File: listPrime.go
package main

import (
	"os"
	"fmt"
	"math"
	"strconv"
)

func IsPrime(value int) bool {
	squareOfValue := int(math.Sqrt(float64(value)))
	for i := 2; i <= squareOfValue; i++ { if value % i == 0 { return false } }
	return true
}

func ListPrime(maxValue int) {
	for i := 2; i <= maxValue; i++ { if IsPrime(i) { fmt.Println(i) } }
}

func main() {
	maxValue := 0
	if len(os.Args) > 1 {
		maxValue, _ = strconv.Atoi(os.Args[1])
	} else { 
		buffer := ""
		fmt.Print("Enter Max Value: ")
		fmt.Scanln(&buffer)
		maxValue, _ = strconv.Atoi(buffer)
	}
	ListPrime(maxValue)
}
